userspace/libsinsp/filter/parser.cpp

```diff
@@ -0,0 +1,572 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <cstring>
+#include <regex>
+#include "parser.h"
+#include "../utils.h"
+#include "../sinsp_exception.h"
+
+using namespace libsinsp::filter;
+
+static const vector<string> unary_ops =
+{ 
+	"exists"
+};
+
+static const vector<string> binary_num_ops = 
+{ 
+	"<=", "<", ">=", ">"
+};
+
+static const vector<string> binary_str_ops =
+{
+	"==", "=", "!=", "glob ", "contains ", "icontains ",
```

Comment:
Why do these have trailing spaces?

Reply:
This is an implementation trick that enforces some operators to be followed by a space. For example:
- `evt.type=open`, `evt.num>=1` are acceptable
- `proc.name containscat` is not acceptable, and should instead be `proc.name contains cat`.

This is already done for common sense in the filters we have, but I don't think this was explicitly enforced before. This only applies for operators composed by simple characters, such as `glob`, `contains`, and so on. We don't have this trouble with special-chars operators such as `=` or `>=`.

Reply:
Ok got it!